py/desitarget/mock/build.py

```diff
@@ -1042,3 +1060,509 @@ def join_targets_truth(mockdir, outdir=None, force=False, comm=None):
         mtl.meta['EXTNAME'] = 'MTL'
         mtl.write(tmpout, overwrite=True, format='fits')
         os.rename(tmpout, out_mtl)
+
+def _get_magnitudes_onebrick(specargs):
+    """Filler function for the multiprocessing."""
+    return get_magnitudes_onebrick(*specargs)
+
+def get_magnitudes_onebrick(target_name, mockformat, thisbrick, brick_info, Magnitudes, dust_dir,
+                         select_targets_function, source_data, rand, log):
+    """Wrapper function to generate magnitudes for all the objects on a single brick."""
+
```

Comment:
This function (`get_magnitudes_onebrick`) contains a significant amount of cut-and-pasted code from the corresponding `get_spectra_onebrick` function (including the debugging plots, extinction coefficients, etc.).   Could you please refactor the common code so that the two functions just contain the code they need.  This will make it *significantly* easier for me to implement the speed-up I have in mind without having to change the same code in two places.